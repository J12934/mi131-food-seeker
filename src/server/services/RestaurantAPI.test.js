/**
 * @jest-environment node
 */

const RestaurantAPI = require('./RestaurantAPI');

describe('Restaurant', () => {
    it('should transform returned API Data', async () => {
        class GraphQLClientMock {
            request = jest.fn(() => {
                return Promise.resolve({
                    search: { business: ['foo', 'bar'] },
                });
            });
        }

        const restaurantAPI = new RestaurantAPI({
            graphQLClient: GraphQLClientMock,
        });

        const results = await restaurantAPI.search({
            term: 'burger',
            location: 'Kiel',
        });

        expect(
            restaurantAPI._graphqlClient.request.mock.calls
        ).toMatchSnapshot();
        expect(results).toMatchSnapshot();
    });
});
