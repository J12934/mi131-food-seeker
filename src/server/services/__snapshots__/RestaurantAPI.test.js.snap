// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Restaurant should transform returned API Data 1`] = `
Array [
  Array [
    "
            query Search($term: String!, $location: String!) {
                search(term: $term, location: $location) {
                    total
                    business {
                        id
                        name
                        rating
                        coordinates {
                            latitude
                            longitude
                        }
                    }
                }
            }",
    Object {
      "location": "Kiel",
      "term": "burger",
    },
  ],
]
`;

exports[`Restaurant should transform returned API Data 2`] = `
Array [
  "foo",
  "bar",
]
`;
